src/Microsoft.ML.TimeSeries/RootCauseAnalyzer.cs

```diff
@@ -10,34 +10,39 @@
 
 namespace Microsoft.ML.TimeSeries
 {
-    public class RootCauseAnalyzer
+    internal class RootCauseAnalyzer
     {
         private static double _anomalyRatioThreshold = 0.5;
         private static double _anomalyDeltaThreshold = 0.95;
         private static double _anomalyPreDeltaThreshold = 2;
 
         private RootCauseLocalizationInput _src;
         private double _beta;
+        private RootCauses _rootCauses;
 
         public RootCauseAnalyzer(RootCauseLocalizationInput src, double beta)
         {
             _src = src;
             _beta = beta;
+            _rootCauses = new RootCauses() { Causes = new List<RootCause>() };
         }
 
         public RootCause Analyze()
+        {
+            return AnalyzeOneLayer(_src).Causes.FirstOrDefault();
+        }
+
+        public RootCauses AnalyzeAll()
```

Comment:
Will something like AnalyzeMultiDimensionalRootCauses be better? AnalyzeAll might also indicate the overall level

Reply:
Yes, I think so